### PR TITLE
=per #18559 TCK tests for highestSequenceNr not reset after journal message deletion

### PR DESCRIPTION
**Pull request by eshepelyuk**
TCK tests for #18559

### CODE REVIEW

**Comment by ktoso** [comment]
OK TO TEST

**Comment by ktoso** [comment]
OK TO TEST

**Comment by ktoso** [comment]
Good tests, I think it also needs to be more explicit in documentation, could you add a bit about this to `persistence.rst`?

**Comment by eshepelyuk** [comment]
Added some notes to peristence docs. Please let me know if need further adjustment.

**Comment by johanandren** [comment]
LGTM

**Comment by ktoso** [comment]
LGTM, please have a look @patriknw as you suggested the test case in the ticket and merge at will.

**Comment by patriknw** [comment]
@eshepelyuk Can you please also add description in `asyncReadHighestSequenceNr` in `AsyncRecovery.scala` and `asyncDeleteMessagesTo` in `AsyncWriteJournal.scala` that the journal must maintain the highest sequence number.

File: akka-persistence-tck/src/main/scala/akka/persistence/journal/JournalSpec.scala [hunk 1]
```diff
@@ -157,6 +157,31 @@ abstract class JournalSpec(config: Config) extends PluginSpec(config) {
...
+      journal ! ReplayMessages(0, Long.MaxValue, Long.MaxValue, pid, receiverProbe.ref)
+      1 to 5 foreach { i ⇒ receiverProbe.expectMsg(replayedMessage(i)) }
+      receiverProbe.expectMsg(RecoverySuccess(highestSequenceNr = 5L))
+
+      journal ! DeleteMessagesTo(pid, Long.MaxValue, receiverProbe.ref)
+      receiverProbe.expectMsg(DeleteMessagesSuccess(Long.MaxValue))
+
+      journal ! ReplayMessages(0, Long.MaxValue, Long.MaxValue, pid, receiverProbe.ref)
+      receiverProbe.expectMsg(RecoverySuccess(highestSequenceNr = 5L))
+    }
```
Comment (patriknw) [inline akka-persistence-tck/src/main/scala/akka/persistence/journal/JournalSpec.scala]: good test, thanks

**Comment by eshepelyuk** [comment]
Added docs to mentioned methods.

File: akka-persistence/src/main/scala/akka/persistence/journal/AsyncWriteJournal.scala [hunk 2]
```diff
@@ -215,6 +215,7 @@ trait AsyncWriteJournal extends Actor with WriteJournalBase with AsyncRecovery {
    * (inclusive).
    *
    * This call is protected with a circuit-breaker.
+   * Message deletion doesn't affect the highest sequence number of messages, journal must maintain the highest sequence number and never decrease it.
```
Comment (ktoso) [inline akka-persistence/src/main/scala/akka/persistence/journal/AsyncWriteJournal.scala]: :+1:

**Comment by patriknw** [comment]
This looks ready. Please squash to one commit and we will merge it.

**Comment by patriknw** [comment]
LGTM

**Comment by eshepelyuk** [comment]
Squash is done

**Comment by patriknw** [comment]
PLS BUILD

**Comment by patriknw** [comment]
@eshepelyuk thanks for a great contribution. There are more persistence issues if you are interested.
https://github.com/akka/akka/issues?q=is%3Aopen+label%3At%3Apersistence+milestone%3A2.4.1
https://github.com/akka/akka/issues?q=is%3Aopen+label%3At%3Apersistence+milestone%3A2.4.x

**Comment by eshepelyuk** [comment]
@patriknw Thank you very much for accepting the PR.
I'd like to mention another small persistence PR #18698 that is not included into any release , but maybe you can review it and provide feedback.

**Comment by patriknw** [comment]
Thanks for the reminder. That is a good one. I'll take a look.

**Pull request merged**